src/jrd/vio.cpp

```diff
@@ -6294,6 +6299,9 @@ static int prepare_update(	thread_db*		tdbb,
 				{
 					tdbb->bumpRelStats(RuntimeStatistics::RECORD_CONFLICTS, relation->rel_id);
 
+					if (writeLockSkipLocked == true)
+						return PREPARE_LOCKERR;
```

Comment:
I'm not sure I like `PREPARE_LOCKERR` meaning somewhat different things depending on the `writeLockSkipLocked` parameter. Maybe we need a different result code e.g. `PREPARE_LOCK` for this particular task?